### PR TITLE
get all esdts roles

### PR DESCRIPTION
**Pull request by miiu96**
- Added a new API route /address/:address that will return all esdts tokens with roles

### CODE REVIEW

File: api/groups/baseAccountsGroup.go [hunk 1]
```diff
@@ -257,6 +258,34 @@ func (group *accountsGroup) getESDTTokenData(c *gin.Context) {
 	c.JSON(http.StatusOK, esdtTokenResponse)
 }
 
+func (group *accountsGroup) getESDTsRoles(c *gin.Context) {
```
Comment (bogdan-rosianu) [inline api/groups/baseAccountsGroup.go:261]: add unit tests

File: process/accountProcessor.go [hunk 2]
```diff
@@ -163,6 +163,35 @@ func (ap *AccountProcessor) GetESDTsWithRole(address string, role string) (*data
 	return nil, ErrSendingRequest
 }
 
+func (ap *AccountProcessor) GetESDTsRoles(address string) (*data.GenericAPIResponse, error) {
```
Comment (bogdan-rosianu) [inline process/accountProcessor.go:166]: don't forget about comments for exported members

File: process/accountProcessor.go [hunk 3]
```diff
@@ -163,6 +163,35 @@ func (ap *AccountProcessor) GetESDTsWithRole(address string, role string) (*data
...
+	observers, err := ap.proc.GetObservers(core.MetachainShardId)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, observer := range observers {
+		apiResponse := data.GenericAPIResponse{}
+		apiPath := AddressPath + address + "esdts/roles"
```
Comment (bogdan-rosianu) [inline process/accountProcessor.go:174]: extract constant. also, bug: there's no `/` between address and `esdts/roles`

File: process/accountProcessor.go (hunk 2 above)
Comment (bogdan-rosianu) [reply process/accountProcessor.go:166]: add unit test

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**